modules/sharedaddy/sharing.js

```diff
@@ -1,364 +1,532 @@
+		}
+	}
 
-				if ( ! ( permalink in WPCOM_sharing_counts ) ) {
-					continue;
-				}
+	// Helper function to remove a node from the DOM.
+	function removeNode( node ) {
+		if ( node && node.parentNode ) {
+			node.parentNode.removeChild( node );
+		}
+	}
 
-				WPCOMSharing.inject_share_count(
-					'sharing-facebook-' + WPCOM_sharing_counts[ permalink ],
-					data[ url ].share.share_count
-				);
-			}
-		},
-		update_pinterest_count: function ( data ) {
-			if ( 'undefined' !== typeof data.count && data.count * 1 > 0 ) {
-				WPCOMSharing.inject_share_count(
-					'sharing-pinterest-' + WPCOM_sharing_counts[ data.url ],
-					data.count
-				);
-			}
-		},
-		inject_share_count: function ( id, count ) {
-			var $share = jQuery( 'a[data-shared=' + id + '] > span' );
-			$share.find( '.share-count' ).remove();
-			$share.append(
-				'<span class="share-count">' + WPCOMSharing.format_count( count ) + '</span>'
-			);
-		},
-		format_count: function ( count ) {
-			if ( count < 1000 ) {
-				return count;
-			}
-			if ( count >= 1000 && count < 10000 ) {
-				return String( count ).substring( 0, 1 ) + 'K+';
-			}
-			return '10K+';
-		},
-		bump_sharing_count_stat: function ( service ) {
-			new Image().src =
-				document.location.protocol +
-				'//pixel.wp.com/g.gif?v=wpcom-no-pv&x_sharing-count-request=' +
-				service +
-				'&r=' +
-				Math.random();
-		},
+	// Helper functions to show/hide a node, and check its status.
+	function hideNode( node ) {
+		if ( node ) {
+			node.style.display = 'none';
+		}
+	}
+
+	function showNode( node ) {
+		if ( node ) {
+			node.style.removeProperty( 'display' );
+		}
+	}
+
+	function isNodeHidden( node ) {
+		return ! node || node.style.display === 'none';
+	}
+
+	// ------------------------------- CLASSES ------------------------------- //
+
+	var PANE_SELECTOR = '.sharing-hidden .inner';
+	var PANE_DATA_ATTR = 'data-sharing-more-button-id';
+
+	// Implements a MoreButton class, which controls the lifecycle and behavior
+	// of a "more" button and its dialog.
+	function MoreButton( buttonEl ) {
+		this.button = buttonEl;
+		this.pane = closest( buttonEl, 'div' ).querySelector( PANE_SELECTOR );
+		this.openedBy = null;
+		this.recentlyOpenedByHover = false;
+
+		MoreButton.instances.push( this );
+		this.pane.setAttribute( PANE_DATA_ATTR, MoreButton.instances.length - 1 );
+
+		this.attachHandlers();
+	}
+
+	// Keep a reference to each instance, so we can get back to it from the DOM.
+	MoreButton.instances = [];
+
+	// Delay time configs.
+	MoreButton.hoverOpenDelay = 200;
+	MoreButton.recentOpenDelay = 400;
+	MoreButton.hoverCloseDelay = 300;
+
+	// Use this to avoid creating new instances for buttons which already have one.
+	MoreButton.instantiateOrReuse = function ( buttonEl ) {
+		var pane = buttonEl && closest( buttonEl, 'div' ).querySelector( PANE_SELECTOR );
+		var paneId = pane && pane.getAttribute( PANE_DATA_ATTR );
+
+		var existingInstance = paneId && MoreButton.instances[ paneId ];
```

Comment:
This will fail to reuse `MoreButton.instances[ 0 ]`.

Below on line 129 you seem to rely on `MoreButton.instances[ null ]` returning undefined, you could do the same here.
```suggestion
		var existingInstance = MoreButton.instances[ paneId ];
```